apps/ios/Shared/Views/NewChat/NewChatView.swift

```diff
@@ -97,6 +97,11 @@ struct NewChatView: View {
             }
             .pickerStyle(.segmented)
             .padding()
+            .onChange(of: $selection.wrappedValue) { opt in
```

Comment:
I think old behavior when swiping from "new link" to "scan link" was to show scan button, and not immediately scanner (camera) itself.. but it's probably ok

Reply:
on one hand seems like a good idea, on the other hand this is more consistent with what happens when you open it directly so not totally sure

Reply:
i would say that if you go there is a lot more likely that you want to scan then to paste the link (since paste is supported everywhere else) and we would be adding an extra tap for the most common action